### DISCUSSION TITLE
Allow unit explosion during conversion

### DISCUSSION THREAD

**Discussion by facetious** [convert-units/convert-units#344] [Ideas]
Most of the time, we want a single number from a single number. This is what this module is primarily designed for and it does very well at this. But what about adding an extension that would allow a set of numbers in response? Specifically, a trailing "next best" rather than resorting to decimals.

**Use case** 
Imagine you have "525601 minutes" and you'd like it to be a little better. You could use the existing tools and get
```
convert(525601).from('minutes').toBest();
// { val: 11.99180926306183, unit: 'month', singular: 'Month', plural: 'Months' }
```
, but that's not better. (Also, sidebar: this calculation is clearly wrong in this module since it should be > 1 year...)

If an array were returned, we could imagine getting:
```
convert(525601).from('minutes').toBest({ explode: true });
// [ { val: 1, unit: 'year', singular: 'Year', plural: 'Years' }, { val: 1, unit: 'minute', singular: 'Minute', plural: 'Minutes' }]
```

I'm not married to the term `explode` here, but naming is hard and I just wanted to get the discussion rolling.

**Comment by Taar** [comment]
The calculation is correct given that it is using `356.25` not `356` as the value for days in a single year. Perhaps the confusion around this could be fixed with a new unit called "calendar year" unit which would instead use `356` but even then, it's still an approximation.

Honestly, `toBest` is already extremely subjective and doesn't work well for all measurement systems. I feel that it should be renamed to `closestUnitToZero` or something like that. With that said, I don't think should sort of functionality should be tacked onto the `toBest` as it would add a lot more complexity to the method. It would be better to support functionality like this as a separate method. However, in my opinion, I don't think `toBest` fits within the scope of this project to begin with and adding more methods that are just a wrapper around the `to` method isn't something I would like to maintain nor entertain implementing at this moment in time. Right now I would like this project to stay focused on the one thing, which is providing an simple way to convert units for those that don't want to write something themselves.
